doc/design/vic-machine/configure.md

```diff
@@ -0,0 +1,182 @@
+	Cpu: 1
+	Memory: 2048 MHZ
+	Debug: 3
+	SSHEnabled: true
+```
+
+- Command Option Format
+
+  The yml format output is more readable for VIC admin, but if users want to recreate their vic-machine create or configure command, it's hard to do based on that format, because there is no one on one mapping between the configuration and command options. 
+
+  Here we'd support another option as "vic-machine inspect --cmd --conf", this will print out the vic-machine create command options minus secret information. 
+
+  The sample output:
+```
+--target                vc.domain.com
+--name                  VCH-1
+--compute-resource      /vcqaDC/host/cls/Resources
+--ops-user              operator
+--image-store           datastore1
+--volume-store          ds://datastore/volumes/default:default
+--volume-store          nfs://userid@host:port/volumes/:nfs
+--bridge-network        bridge
+--bridge-network-range  172.16.0.0/12
+...
+```
+
+Based on this output, it's easy to modify existing configuration to configure command options
+
+Note: Special character might be escaped based on current OS platform. But even with this, the command options cannot be copied to other platform.
+
+- Subcommand Inspect (corresponding to configure subcommand option)
+  The configure subcommand option might change command option based on each different sub command requirement, to get current configuration, we'll need to have command inspect for each subcommand. So the option is to extend vic-machine inspect command to include sub command as well.
+  For example, ```vic-machine inspect bridge-network``` will print out current configuration in the command format, to make sure user can easily generate their configure command based on that.
+
+Note: this is not needed atm, because we're not doing sub configure command
+
+## Update In Progress Status
+Same to vic-machine upgrade, concurrent configure requests for same VCH is not allowed. vic-machine should be able to detect if one VCH is running configure/upgrade. If true, return useful error message. And also should not leave that configure flag in there after configure is stopped, or even interrupted.
+
+Issue https://github.com/vmware/vic/issues/4069 is tracking this problem.
+
+## Backward Compatibility
+It's possible that after vic engine released new version, user is trying to configure older version's VCH with newer vic-machine.
+
+There are few options:
+- Run configure using same version's vic-machine (this option should technically always work)
+
+  For any customer that does not want to change vic-machine version, they need to run configure using same version's vic-machine.
+
+- Combine upgrade process into configure
+
+  Upgrade is actually one special kind of configure, technically it's easy to have upgrade run together with configure. User might want to configure VCH, and also want to upgrade it after or before configure. In that case, a combined configure and upgrade command can provide a better user experience.
```

Comment:
Definitely. There is also value in being able to rollback an incorrect configure as noted below.